src/main/java/com/synopsys/integration/jenkins/detect/service/strategy/DetectAirGapJarStrategy.java

```diff
@@ -0,0 +1,137 @@
+
+public class DetectAirGapJarStrategy extends DetectExecutionStrategy {
+    private static final String DETECT_JAR_PREFIX = "synopsys-detect-";
+    private static final String DETECT_JAR_SUFFIX = ".jar";
+
+    private final JenkinsIntLogger logger;
+    private final IntEnvironmentVariables intEnvironmentVariables;
+    private final String remoteJdkHome;
+    private final JenkinsConfigService jenkinsConfigService;
+    private final AirGapDownloadStrategy airGapDownloadStrategy;
+
+    public DetectAirGapJarStrategy(JenkinsIntLogger logger, IntEnvironmentVariables intEnvironmentVariables, String remoteJdkHome, JenkinsConfigService jenkinsConfigService, AirGapDownloadStrategy airGapDownloadStrategy) {
+        this.logger = logger;
+        this.intEnvironmentVariables = intEnvironmentVariables;
+        this.remoteJdkHome = remoteJdkHome;
+        this.jenkinsConfigService = jenkinsConfigService;
+        this.airGapDownloadStrategy = airGapDownloadStrategy;
+    }
+
+    @Override
+    public Function<String, String> getArgumentEscaper() {
+        return Function.identity();
+    }
+
+    @Override
+    public MasterToSlaveCallable<ArrayList<String>, IntegrationException> getSetupCallable() {
+        return new SetupCallableImpl(logger, intEnvironmentVariables.getVariables(), remoteJdkHome, jenkinsConfigService, airGapDownloadStrategy);
+    }
+
+    public static class SetupCallableImpl extends MasterToSlaveCallable<ArrayList<String>, IntegrationException> {
+        private static final long serialVersionUID = -8326836838838706367L;
+
+        private final JenkinsIntLogger logger;
+        private final Map<String, String> environmentVariables;
+        private final String remoteJdkHome;
+        private final JenkinsConfigService jenkinsConfigService;
+        private final AirGapDownloadStrategy airGapDownloadStrategy;
+
+        public SetupCallableImpl(JenkinsIntLogger logger, Map<String, String> environmentVariables, String remoteJdkHome, JenkinsConfigService jenkinsConfigService, AirGapDownloadStrategy airGapDownloadStrategy) {
+            this.logger = logger;
+            this.environmentVariables = environmentVariables;
+            this.remoteJdkHome = remoteJdkHome;
+            this.jenkinsConfigService = jenkinsConfigService;
+            this.airGapDownloadStrategy = airGapDownloadStrategy;
+        }
+
+        @Override
+        public ArrayList<String> call() throws DetectJenkinsException {
+            String airGapJarPath = getOrDownloadAirGapJar();
+            RemoteJavaService remoteJavaService = new RemoteJavaService(logger, remoteJdkHome, environmentVariables);
+            String javaExecutablePath = remoteJavaService.calculateJavaExecutablePath();
+
+            logger.info("Detect configured: " + airGapJarPath);
+
+            return new ArrayList<>(Arrays.asList(javaExecutablePath, "-jar", airGapJarPath));
+        }
+
+        private String getOrDownloadAirGapJar() throws DetectJenkinsException {
```

Comment:
I wonder if it might not be better to collapse these private methods into the main call() method, since it seems like we do them roughly in order and splitting them off just makes it a little tricky to see what happens when. Do you have any compelling reason why we shouldn't?

Reply:
No reason. Just trying to break code into logical chunks for future supportability.

Reply:
Looking over this code, I think leaving the code in logical chunks as it currently is makes it easier to understand and follow the flow. If combining this is something you feel strongly about maybe we can jump on a call to discuss more.

Reply:
If you're happy with it, I don't have any objections-- just wanted to challenge it and make sure you had a good reason for it, since private methods can often proliferate.